x-pack/legacy/plugins/siem/public/containers/detection_engine/signals/api.test.ts

```diff
@@ -0,0 +1,291 @@
+  createSignalIndex,
+} from './api';
+import { ToasterErrors } from '../../../components/ml/api/throw_if_not_ok';
+import { PostSignalError, SignalIndexError } from './types';
+
+const abortCtrl = new AbortController();
+const mockKibanaServices = KibanaServices.get as jest.Mock;
+jest.mock('../../../lib/kibana');
+
+const mockfetchSuccess = (body: unknown, fetchMock?: jest.Mock) => {
+  if (fetchMock) {
+    mockKibanaServices.mockImplementation(() => ({
+      http: {
+        fetch: fetchMock,
+      },
+    }));
+  } else {
+    mockKibanaServices.mockImplementation(() => ({
+      http: {
+        fetch: () => ({
+          response: {
+            ok: true,
+            message: 'success',
+            text: 'success',
+          },
+          body,
+        }),
+      },
+    }));
+  }
+};
+
+const mockfetchError = () => {
+  mockKibanaServices.mockImplementation(() => ({
+    http: {
+      fetch: () => ({
+        response: {
+          ok: false,
+          text: () =>
+            JSON.stringify({
+              message: 'super mega error, it is not that bad',
+            }),
+        },
+        body: null,
+      }),
+    },
+  }));
+};
+
+describe('Detections Signals API', () => {
+  const fetchMock = jest.fn();
+
+  describe('fetchQuerySignals', () => {
+    beforeEach(() => {
+      mockKibanaServices.mockClear();
+      fetchMock.mockClear();
+      fetchMock.mockImplementation(() => ({
+        response: {
+          ok: true,
+          message: 'success',
+          text: 'success',
+        },
+        body: signalsMock,
+      }));
+    });
+    test('check parameter url, body', async () => {
+      mockfetchSuccess(null, fetchMock);
+
+      await fetchQuerySignals({ query: mockSignalsQuery, signal: abortCtrl.signal });
+      expect(fetchMock).toHaveBeenCalledWith('/api/detection_engine/signals/search', {
+        asResponse: true,
+        body:
+          '{"aggs":{"signalsByGrouping":{"terms":{"field":"signal.rule.risk_score","missing":"All others","order":{"_count":"desc"},"size":10},"aggs":{"signals":{"date_histogram":{"field":"@timestamp","fixed_interval":"81000000ms","min_doc_count":0,"extended_bounds":{"min":1579644343954,"max":1582236343955}}}}}},"query":{"bool":{"filter":[{"bool":{"must":[],"filter":[{"match_all":{}}],"should":[],"must_not":[]}},{"range":{"@timestamp":{"gte":1579644343954,"lte":1582236343955}}}]}}}',
+        method: 'POST',
+        signal: abortCtrl.signal,
+      });
+    });
+    test('happy path', async () => {
+      mockfetchSuccess(signalsMock);
+      const signalsResp = await fetchQuerySignals({
+        query: mockSignalsQuery,
+        signal: abortCtrl.signal,
+      });
+      expect(signalsResp).toEqual(signalsMock);
+    });
+    test('unhappy path', async () => {
+      mockfetchError();
+      try {
+        await fetchQuerySignals({ query: mockSignalsQuery, signal: abortCtrl.signal });
+      } catch (exp) {
+        expect(exp).toBeInstanceOf(ToasterErrors);
```

Comment:
https://github.com/elastic/kibana/commit/625aa5f2c24d3b306f2213ce6af5ba99d0f73e86 also uncovered some potential false positives on these try/catch tests: if `fetchQuerySignals` doesn't throw an error, we don't make any assertions.